test/e2e/traceflow_test.go

```diff
@@ -184,6 +336,24 @@ func TestTraceflow(t *testing.T) {
 		}
 	}()
 
+	if intraNodeUDPDstIPTraceflow, err = data.crdClient.OpsV1alpha1().Traceflows().Create(context.TODO(), intraNodeUDPDstIPTraceflow, v1.CreateOptions{}); err != nil {
+		t.Fatalf("Error when creating traceflow: %v", err)
+	}
+	defer func() {
+		if err := data.crdClient.OpsV1alpha1().Traceflows().Delete(context.TODO(), intraNodeUDPDstIPTraceflow.Name, v1.DeleteOptions{}); err != nil {
+			t.Fatalf("Error when deleting traceflow: %v", err)
+		}
+	}()
+
+	if interNodeICMPDstIPTraceflow, err = data.crdClient.OpsV1alpha1().Traceflows().Create(context.TODO(), interNodeICMPDstIPTraceflow, v1.CreateOptions{}); err != nil {
+		t.Fatalf("Error when creating traceflow: %v", err)
+	}
+	defer func() {
+		if err := data.crdClient.OpsV1alpha1().Traceflows().Delete(context.TODO(), interNodeICMPDstIPTraceflow.Name, v1.DeleteOptions{}); err != nil {
+			t.Fatalf("Error when deleting traceflow: %v", err)
+		}
+	}()
+
```

Comment:
regarding the `wait.Poll` validation "loop" below: I know it wasn't introduced in this PR, but it seems like a weird way of doing things. If not all traceflow CRDs finish at the same time, you may run the validation checks on the same CRD more than once.
I am also not sure that this if statement:
```go
if done == len(testcases) {
    return true, nil
}
```
actually serves a purpose, since you never actually break from the loop: you either return from the function early or the statement is guaranteed to be true.

Maybe we could have something like this:
```go
    waitForTraceflow := func(tfName string) *v1alpha1.Traceflow, err {
        wait.PollImmediate(...)
    } 
    for _, tc := range testcases {
        tc := tc // capture range variable
        t.Run(tc.Name, func(t *testing.T) { // having a name would be nice (e.g. "intraNodeTraceflow")
            // can these tests be run in parallel? If yes, we can call t.Parallel()
            tf, err := waitForTraceflow(tc.tf.Name)
            // check err
            // run validation checks
        })
    }
```
Anyway let me know what you think.

Reply:
Yes, with the old design, one traceflow may be validated for several times before all traceflow validation pass. It is not efficient enough.

Thank you for the great suggestion and I use t.Run() now! Please note that I wrap t.Run() into a function in order to use WaitGroup. These tests should be parallel but without WaitGroup, pods and networkpolicy cleanup will be conducted before traceflow finish. I didn't find a way to add WaitGroup to t.Run() so I wrapped it. Please share your idea if there is a better way.

Reply:
The go documentation for the `testing` package suggests using a nested `Run` call.

To quote the [documentation](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks):
> Run does not return until parallel subtests have completed, providing a way to clean up after a group of parallel tests:
```
func TestTeardownParallel(t *testing.T) {
    // This Run will not return until the parallel tests finish.
    t.Run("group", func(t *testing.T) {
        t.Run("Test1", parallelTest1)
        t.Run("Test2", parallelTest2)
        t.Run("Test3", parallelTest3)
    })
    // <tear-down code>
}
```

Reply:
Thank you! Updated.